llvm/include/llvm/ProfileData/MemProf.h

```diff
@@ -1166,6 +1171,27 @@ template <typename FrameIdTy> class CallStackRadixTreeBuilder {
 } // namespace memprof
 
 namespace yaml {
+template <> struct ScalarTraits<memprof::GUIDHex64> {
+  static void output(const memprof::GUIDHex64 &Val, void *, raw_ostream &Out) {
+    // Print GUID as a 16-digit hexadecimal number.
+    Out << format("0x%016" PRIx64, (uint64_t)Val);
+  }
+  static StringRef input(StringRef Scalar, void *, memprof::GUIDHex64 &Val) {
+    uint64_t Num;
+    if (Scalar.starts_with_insensitive("0x")) {
+      // Accept hexadecimal numbers starting with 0x or 0X.
+      if (Scalar.getAsInteger(0, Num))
+        return "invalid hex64 number";
+      Val = Num;
+    } else {
+      // Otherwise, treat the input as a string containing a function name.
+      Val = memprof::IndexedMemProfRecord::getGUID(Scalar);
+    }
+    return StringRef();
```

Comment:
Should we fail or complain somehow instead of returning the empty string?
It seems that now we only accept string or hex formatted numbers so the old guid format as int will silently return empty.

Reply:
Returning the empty string indicates a success here.

If a decimal number come in, we would treat that as a function name.  I could extend the logic to accept decimal numbers to be a little more user friendly.

Reply:
I think it's fine to disallow decimals as long as we inform the user.

Reply:
I've added code to reject decimal numbers.